cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java

```diff
@@ -110,10 +120,21 @@ public ScheduleTaskRunner(Store store, ProgramLifecycleService lifecycleService,
                                       Map<String, String> userArgs) throws Exception {
     ProgramRuntimeService.RuntimeInfo runtimeInfo;
     try {
+      // if the program has a namespace user configured then set that user in the security request context.
+      // See: CDAP-7396
+      String nsPrincipal = namespaceQueryAdmin.get(id.getNamespace()).getConfig().getPrincipal();
+      if (nsPrincipal != null) {
+        SecurityRequestContext.setUserId(new KerberosName(nsPrincipal).getServiceName());
```

Comment:
``NamespaceConfig`` is just static data that is not dependent on the cluster configuration, right? What happens if someone sets the principal, but does not enable security/kerberos on the cluster?

Also, this only really matters when security is enabled, so it would be better to have the check as ``nsPrincipal != null && SecurityUtil.isKerberosEnabled(cConf)``?